cmd/consumer/main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/config"
+	"github.com/emmaLP/gs-software-onboarding/internal/consumer"
+	"go.uber.org/zap"
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger, err := configureLogger()
+	if err != nil {
+		log.Fatal("Failed to configure the logger", err)
+	}
+
+	configuration, err := config.LoadConfig(".")
+	if err != nil {
+		logger.Fatal("Failed to load config", zap.Error(err))
+	}
+
+	if err := consumer.ConfigureCron(ctx, logger, configuration); err != nil {
+		logger.Fatal("Failed to configure the cron", zap.Error(err))
+	}
+}
+
+func configureLogger() (*zap.Logger, error) {
+	logger, err := zap.NewProduction()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create logger: %w", err)
+	}
+	defer func() {
```

Comment:
`logger.Sync` ensures anything stored in the logger's buffer is flushed prior to the program exiting ([see more](https://pkg.go.dev/go.uber.org/zap#Logger.Sync)).

Currently, this `defer` will call `logger.Sync()` when the `configureLogger` function returns rather than when the program exits. To ensure the `logger.Sync` is called when `main` exits, It could return a clean up function from this `configureLogger` to defer call in `main. Alternatively, another option is to move this code into `main` as there isn't much to it and it's pretty typical to have this kind of setup in `main`.

Reply:
Good spot. Will correct